sources/packages/web/src/views/institution/OfferingsUpload.vue

```diff
@@ -0,0 +1,316 @@
+                      {{ warning.warningMessage }}
+                    </li>
+                  </ul>
+                </div>
+              </template></Column
+            >
+          </DataTable>
+        </content-group>
+      </content-group>
+    </content-group>
+  </full-page-container>
+</template>
+
+<script lang="ts">
+import { ref, computed } from "vue";
+import {
+  OfferingsUploadBulkInsert,
+  DEFAULT_PAGE_LIMIT,
+  PAGINATION_LIST,
+  OfferingStatus,
+  BannerTypes,
+  VForm,
+  FileInputFile,
+} from "@/types";
+import { EducationProgramOfferingService } from "@/services/EducationProgramOfferingService";
+import StatusChipOffering from "@/components/generic/StatusChipOffering.vue";
+import { useSnackBar } from "@/composables";
+import { FileUploadProgressEventArgs } from "@/services/http/common/FileUploadProgressEvent";
+
+const ACCEPTED_FILE_TYPE = "text/csv";
+const MAX_OFFERING_UPLOAD_SIZE = 4194304;
+
+export default {
+  components: {
+    StatusChipOffering,
+  },
+  setup() {
+    const snackBar = useSnackBar();
+    const loading = ref(false);
+    // Only one will be used but the component allows multiple.
+    const offeringFiles = ref([] as FileInputFile[]);
+    // Possible errors and warnings received upon file upload.
+    const validationResults = ref([] as OfferingsUploadBulkInsert[]);
+    const uploadForm = ref({} as VForm);
+    const uploadProgress = ref({} as FileUploadProgressEventArgs);
+    const showValidationSummary = ref(false);
+    // Workaround to reset the file upload component to its original state.
+    // It is apparently a vuetify beta issue. It can be removed once there is a
+    // better way to force the component to reset its state.
+    const csvFileUploadKey = ref(0);
+    // Specific error message to be display the error that occurs when the file upload
+    // has a file selected and the user changes its contents (net::ERR_UPLOAD_FILE_CHANGED).
+    const showPossibleFileChangeError = ref(false);
+
+    const uploadFile = async (validationOnly: boolean) => {
+      const validationResult = await uploadForm.value.validate();
+      if (!validationResult.valid) {
+        return;
+      }
+      showPossibleFileChangeError.value = false;
+      try {
+        loading.value = true;
+        const [fileToUpload] = offeringFiles.value;
+        const uploadResults =
+          await EducationProgramOfferingService.shared.offeringBulkInsert(
+            fileToUpload,
+            validationOnly,
+            (progressEvent: FileUploadProgressEventArgs) => {
+              uploadProgress.value = progressEvent;
+            },
+          );
+        if (uploadResults.length) {
+          validationResults.value = uploadResults;
+          showValidationSummary.value = true;
+        } else {
+          resetForm();
+          snackBar.success("File successfully upload.");
+        }
+      } catch (error: unknown) {
+        if (error instanceof Error && error.message === "Network Error") {
```

Comment:
`"Network Error"`?

Reply:
It was what I was able to do in order to try to identify the "file changed error".
The message in the error object does not contain anything related to it, as far as I could check.
If you see the error message displayed to the user you can notice that it also has a generic message.

"An error occurred during the file upload. Reasons can include a file modification after the file was already selected or an unexpected network failure. Please select the file and try again."